texar/torch/data/data/data_iterators.py

```diff
@@ -344,100 +346,224 @@ def __len__(self):
         raise TypeError("DynamicBatchSampler does not support __len__")
 
 
-class _DataLoaderIter(torch_DataLoaderIter):  # pylint: disable=abstract-method
-    r"""Iterates once over the DataLoader's dataset. This is almost identical
-    to PyTorch :class:`torch.utils.data.dataloader._DataLoaderIter`, except
-    that we check `allow_smaller_final_batch` here. This is because using
-    `drop_last` in :class:`~torch.utils.data.sampler.BatchSampler` would cause
-    the dataset to not load/process/cache certain elements from the final batch,
-    which complicates the already complex logic.
-    """
+# pylint: disable=ungrouped-imports
```

Comment:
I see that we use different logic for data iterators in `pytorch 1.2`. Can we write comments here explaining briefly what changes led us to do this? This will be helpful in future releases of pytorch.

Reply:
I agree. Will add comments.